planning/behavior_path_start_planner_module/src/start_planner_module.cpp

```diff
@@ -93,26 +93,56 @@ void StartPlannerModule::onFreespacePlannerTimer()
 {
   const ScopedFlag flag(is_freespace_planner_cb_running_);
 
-  if (getCurrentStatus() == ModuleStatus::IDLE) {
+  std::shared_ptr<const PlannerData> local_planner_data{nullptr};
+  std::optional<ModuleStatus> current_status_opt{std::nullopt};
+  std::optional<StartPlannerParameters> parameters_opt{std::nullopt};
+  std::optional<PullOutStatus> pull_out_status_opt{std::nullopt};
+  bool is_stopped;
+
+  // begin of critical section
+  {
+    std::lock_guard<std::mutex> guard(sp_planner_data_mutex_);
+    if (sp_planner_data_) {
+      const auto & sp_planner_data = sp_planner_data_.value();
+      local_planner_data = std::make_shared<PlannerData>(sp_planner_data.planner_data);
+      current_status_opt = sp_planner_data.current_status;
+      parameters_opt = sp_planner_data.parameters;
+      pull_out_status_opt = sp_planner_data.main_thread_pull_out_status;
+      is_stopped = sp_planner_data.is_stopped;
+    }
+  }
+  // end of critical section
+  if (!local_planner_data || !current_status_opt || !parameters_opt || !pull_out_status_opt) {
     return;
   }
 
-  if (!planner_data_) {
+  const auto & current_status = current_status_opt.value();
+  const auto & parameters = parameters_opt.value();
+  const auto & pull_out_status = pull_out_status_opt.value();
+
+  if (current_status == ModuleStatus::IDLE) {
     return;
   }
 
-  if (!planner_data_->costmap) {
+  if (!local_planner_data->costmap) {
     return;
   }
 
   const bool is_new_costmap =
-    (clock_->now() - planner_data_->costmap->header.stamp).seconds() < 1.0;
+    (clock_->now() - local_planner_data->costmap->header.stamp).seconds() < 1.0;
   if (!is_new_costmap) {
     return;
   }
 
-  if (isStuck()) {
```

Comment:
This method is only called here, I suggest either modifying the method to accept is_stopped , pull_out_status.planner_type and pull_out_status.found_pull_out_path

and call
` isStuck( is_stopped , pull_out_status.planner_type ,pull_out_status.found_pull_out_path) `

Alternatively, the isStuck() method can be made into a lambda, inside this same function. 

It can also be deleted altogether, but I feel it is more readable if a method called is_ego_stuck() is called here instead of directly checking for 3 different  conditions on this if()

Reply:
fixed in https://github.com/autowarefoundation/autoware.universe/pull/6779/commits/f12bd11c5b8c4b777bff93ef874d2069c0cc9495